### PR TITLE
flux-overlay: improve timeouts, hostname handling

### PR DESCRIPTION
**Pull request by garlick**
This fixes a couple of issues with `flux overlay` identified during testing on fluke yesterday.

### CODE REVIEW

**Comment by garlick** [comment]
Oops posted prematurely.   I should tack on a fix for #4090 to this PR.

**Comment by grondo** [comment]
Oh, I really like the rename of `gethostbyrank` to `flux overlay lookup`.

I know you aren't quite done with this PR, but I did have one quick question.

It seems like on production clusters, disconnected brokers may be a common case. Does a 5s timeout for `flux overlay status` seem excessive, given that (if I'm correct about the common case) this will be the minimum time to run the command? Would it hurt to add a much smaller timeout by default (like maybe 200ms or less), and you already have the `--timeout` option to extend it? (Caveat: I'm probably missing some subtlety here where a 5s timeout is reasonable, and I apologize for bringing it up if so)

**Comment by garlick** [comment]
The timeout really only comes into play with unresponsive nodes that have
not disconnected.  I kind of picked 5s out of nowhere though, and it is a
bit long.  Maybe a few hundred ms would be better, given that there is an
override in case that turns out to be too small.

**Comment by garlick** [comment]
OK this is ready for a review.   I dropped the default `flux overlay status` timeout to 500ms and added the fix for #4090.

File: src/cmd/builtin/overlay.c [hunk 1]
```diff
@@ -39,7 +41,7 @@ static struct optparse_option status_opts[] = {
                " 2=show round-trip RPC times."
     },
     { .name = "timeout", .key = 't', .has_arg = 1, .arginfo = "FSD",
-      .usage = "Set RPC timeout (default none)",
+      .usage = "Set RPC timeout, 0=disable (default 5s)",
```
Comment (grondo) [inline src/cmd/builtin/overlay.c:44]: Update default

**Review by grondo — APPROVED** [review]
This looks good to me. The `flux overlay lookup` command turned out nicely.

Just one small nit called out inline.

File: src/cmd/builtin/overlay.c (hunk 1 above)
Comment (garlick) [reply src/cmd/builtin/overlay.c]: Good catch!  Forcing a push with that change and setting MWP.

**Pull request merged**